### PR TITLE
boost: remove boost submodule

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
tested "make-dist" locally by running it and inspecting it using `tar tjvf ceph-12.1.2-1337-g656cbeba1e.tar.bz2`.

File: make-dist [hunk 1]
```diff
@@ -20,6 +20,27 @@ if ! git submodule sync || ! git submodule update $force --init --recursive; the
...
 fi
 
+download_boost() {
+    boost_version=$1
+    boost_md5=$2
+    boost_version_underscore=$(echo $boost_version | sed 's/\./_/g')
+    boost_fname=boost_${boost_version_underscore}.tar.bz2
+    echo "downloading boost_$boost_version..."
+    wget --no-verbose -O $boost_fname \
+         https://downloads.sourceforge.net/project/boost/boost/$boost_version/$boost_fname
```
Comment (liewegas) [inline make-dist]: Should we stash a copy of this on download.ceph.com or something?

Comment (tchaikov) [reply make-dist]: @alfredodeza do we use a cache enabled http proxy in our CI infrastructure?

Comment (alfredodeza) [reply make-dist]: we don't have any kind of caching in our CI

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2017-09-02-0015/

**Pull request merged**